core/src/main/java/com/segment/analytics/kotlin/core/Telemetry.kt

```diff
@@ -334,7 +351,7 @@ object Telemetry: Subscriber {
     private suspend fun systemUpdate(system: com.segment.analytics.kotlin.core.System) {
         system.settings?.let { settings ->
             settings.metrics["sampleRate"]?.jsonPrimitive?.double?.let {
-                sampleRate = it
+                sampleRate.set(it)
```

Comment:
if use sovran, sampleRate does not have to be atomic, since sovran serialize the task in the same thread. same thing applies to `enable` and `started`